src/services/AuthorizationRequestProcessor.ts

```diff
@@ -0,0 +1,87 @@
+import { Metrics, MetricUnits } from "@aws-lambda-powertools/metrics";
+import { Logger } from "@aws-lambda-powertools/logger";
+import { randomUUID } from "crypto";
+import { BavService } from "./BavService";
+import { AuthSessionState } from "../models/enums/AuthSessionState";
+import { HttpCodesEnum } from "../models/enums/HttpCodesEnum";
+import { MessageCodes } from "../models/enums/MessageCodes";
+import { TxmaEventNames } from "../models/enums/TxmaEvents";
+import { EnvironmentVariables } from "../utils/Constants";
+import { absoluteTimeNow } from "../utils/DateTimeUtils";
+import { createDynamoDbClient } from "../utils/DynamoDBFactory";
+import { checkEnvironmentVariable } from "../utils/EnvironmentVariables";
+import { Response } from "../utils/Response";
+import { buildCoreEventFields } from "../utils/TxmaEvent";
+
+export class AuthorizationRequestProcessor {
+	private static instance: AuthorizationRequestProcessor;
+
+	private readonly logger: Logger;
+
+	private readonly metrics: Metrics;
+
+	private readonly issuer: string;
+
+	private readonly txmaQueueUrl: string;
+
+	private readonly BavService: BavService;
+
+	constructor(logger: Logger, metrics: Metrics) {
+		this.issuer = checkEnvironmentVariable(EnvironmentVariables.ISSUER, logger);
+		this.txmaQueueUrl = checkEnvironmentVariable(EnvironmentVariables.TXMA_QUEUE_URL, logger);
+  	const sessionTableName: string = checkEnvironmentVariable(EnvironmentVariables.SESSION_TABLE, logger);
+
+		this.logger = logger;
+		this.metrics = metrics;
+		this.BavService = BavService.getInstance(sessionTableName, this.logger, createDynamoDbClient());
+	}
+
+	static getInstance(logger: Logger, metrics: Metrics): AuthorizationRequestProcessor {
+		if (!AuthorizationRequestProcessor.instance) {
+			AuthorizationRequestProcessor.instance = new AuthorizationRequestProcessor(logger, metrics);
+		}
+		return AuthorizationRequestProcessor.instance;
+	}
+
+	async processRequest(sessionId: string): Promise<Response> {
+		const session = await this.BavService.getSessionById(sessionId);
+
+		if (!session) {
+			this.logger.error("No session found for session id", {
+				messageCode: MessageCodes.SESSION_NOT_FOUND,
+			});
+			return new Response(HttpCodesEnum.UNAUTHORIZED, `No session found with the session id: ${sessionId}`);
+		}
+
+		this.logger.appendKeys({ govuk_signin_journey_id: session.clientSessionId });
+		this.metrics.addMetric("found session", MetricUnits.Count, 1);
+
+		if (session.authSessionState !== AuthSessionState.BAV_DATA_RECEIVED) {
```

Comment:
Do we also wanna allow BAV_AUTH_CODE_ISSUED in here similar to F2F when we had multiple clicks / back clicks on browser

Reply:
Similar to F2F if state is BAV_AUTH_CODE_ISSUED then instead of generating a new authorizationCode we just return the authorizationCode already saved to the session Record

Reply:
CIC sorry not F2F : https://github.com/govuk-one-login/ipv-cri-cic-api/blob/main/src/services/AuthorizationRequestProcessor.ts#L68

Reply:
Yeah I missed this in the ACs however - they say 

> THEN the lambda will generate a a NEW authorization code and saves this to the SessionTable

This is different to F2F, which is correct?